src/App.js

```diff
@@ -1,85 +1,231 @@
+    });
+  }
+
+  addNewTodo = (event) => {
+    if (event.key === 'Enter'
+      && this.state.title.trim() !== '') {
+      this.setState(state => ({
+        todos: [
+          ...state.todos,
+          {
+            id: state.id + 1,
+            title: state.title,
+            completed: false,
+          }],
+        id: state.id + 1,
+        title: '',
+      }));
+    }
+  }
+
+  deleteTodo = ({ target }) => {
+    const todoId = this.state.todos.findIndex(item => item.id === +target.id);
+
+    this.setState((state) => {
+      const remainingTodos = [...state.todos];
+
+      remainingTodos.splice(todoId, 1);
+
+      return (
+        {
+          todos: [...remainingTodos],
+        }
+      );
+    });
+  }
+
+  changeStatus = (id) => {
+    this.setState(state => ({
+      todos: state.todos.map((todo) => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            completed: !todo.completed,
+          };
+        }
+
+        return todo;
+      }),
+    }));
+  }
+
+  filterAll = () => {
+    this.setState({
+      filter: 'all',
+      isAllBtnSelected: true,
+      isActiveBtnSelected: false,
+      isCompletedBtnSelected: false,
+    });
+  }
+
+  filterActive = () => {
+    this.setState({
+      filter: 'active',
+      isAllBtnSelected: false,
+      isActiveBtnSelected: true,
+      isCompletedBtnSelected: false,
+    });
+  }
+
+  filterCompleted = () => {
+    this.setState({
+      filter: 'completed',
+      isAllBtnSelected: false,
+      isActiveBtnSelected: false,
+      isCompletedBtnSelected: true,
+    });
+  }
+
+  clearCompleted = () => {
+    this.setState(state => ({
+      todos: state.todos.filter(todo => todo.completed === false),
+    }));
+  }
+
+  selectAll = () => {
+    this.setState(state => ({
+      isAllSelected: !state.isAllSelected,
+      todos: state.todos.map(todo => ({
+        ...todo,
+        completed: !state.isAllSelected,
+      })),
+    }));
+  }
 
-      <footer className="footer">
-        <span className="todo-count">
-          3 items left
-        </span>
-
-        <ul className="filters">
-          <li>
-            <a href="#/" className="selected">All</a>
-          </li>
-
-          <li>
-            <a href="#/active">Active</a>
-          </li>
-
-          <li>
-            <a href="#/completed">Completed</a>
-          </li>
-        </ul>
-
-        <button type="button" className="clear-completed">
-          Clear completed
-        </button>
-      </footer>
-    </section>
-  );
+  render() {
+    const { todos,
+      title,
+      filter,
+      isAllSelected,
+      isActiveBtnSelected,
+      isAllBtnSelected,
+      isCompletedBtnSelected } = this.state;
+
+    let currentTodos = [...todos];
+
+    const completedTodos = [];
+    const activeTodos = [];
+
+    todos.forEach((todo) => {
+      if (todo.completed) {
+        completedTodos.push(todo);
+      } else {
+        activeTodos.push(todo);
+      }
+    });
```

Comment:
don't make two separate arrays, just check what tab is chosen and make one array with filtered todos